### DISCUSSION TITLE
Tying payment to online order and customer?

### DISCUSSION THREAD

**Discussion by mckhames** [weareseeed/react-square-web-payments-sdk#6] [Help] [answered]
Thank you so much for developing this awesome module! I have a question regarding its implementation:

I'm building an online takeout system for a Food & Bev client. What checkout flow would you recommend for tying the payment to the customer's account + the customer's order? To be honest, I'm a bit confused about the difference between "processing" the payment, which if I understand correctly, this module assists greatly with, and "creating" a payment tied to both a "customer_id" and "order_id".

We're running NextJS + Firebase Auth + Firestore. To place an order, users will have to create an acct or sign in with Firebase Auth, and Firestore will hold their square customer_id and order/payments history. So once we've created the order via the Orders API, how do we tie this payment to the customer's account/loyalty account?

Sorry if this is too broad a question to ask here. Totally understand if that's the case.

Thanks again for your amazing contributions!

**Comment by rsaer** [comment, marked as answer]
Hi, this is from square, the Orders API Create endpoint has a customer_id field they can populate, and then they can pass that Order ID to v2/payments.

v2/payments also has a customer ID field https://developer.squareup.com/reference/square/payments-api/create-payment

https://developer.squareup.com/reference/square/orders-api/create-order

Creates a new order that can include information about products for purchase and settings to apply to the purchase.

Let us know if you have any other questions

Also you can get support directly from square (if is not related to our SDK) here 
https://docs.google.com/forms/d/e/1FAIpQLSfAZGIEZoNs-XryKqUoW3atFQHdQw5UqXLMOVPq3V4DEq-AJw/viewform?sf76057089=1#response=ACYDBNj5LFgPy8Tcac2gSgv_IjXvgWsPy2CO2xTXwnc0OSSxCvWFgc7SCDHvVQ

Or check the square developer forums
https://developer.squareup.com/forums/
